src/lib/x509/certstor_flatfile/certstor_flatfile.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+* Certificate Store
+* (C) 1999-2019 Jack Lloyd
+* (C) 2019      Patrick Schmidt
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/build.h>
+
+#include <botan/certstor_flatfile.h>
+#include <botan/data_src.h>
+#include <botan/pem.h>
+
+namespace Botan {
+namespace {
+std::vector<secure_vector<uint8_t>> decode_all(DataSource& source)
+   {
+   std::vector<secure_vector<uint8_t>> pems;
+
+   while(!source.end_of_data())
+      {
+      std::string label;
+      try
+         {
+         pems.push_back(PEM_Code::decode(source, label));
+         }
+      catch(const Decoding_Error&) {}
+      }
+
+   return pems;
+   }
+}
+
+Flatfile_Certificate_Store::Flatfile_Certificate_Store()
```

Comment:
This constructor isn't useful, since there is no way to add certs. All you get is a certificate store that does nothing. Instead what it should do is,

```
#if defined(BOTAN_SYSTEM_CERT_BUNDLE)
   // load the certs
#else 
  throw Invalid_State("No default cert bundle located at build time");
#endif 
```

This obviates any need for `Certificate_Store_Linux`.

Reply:
Or, better, remove this constructor, and then applications use the macro value directly `Flatfile_Certificate_Store system_roots(BOTAN_SYSTEM_CERT_BUNDLE)`